src/account/kernel/api/installModule.ts

```diff
@@ -0,0 +1,105 @@
+import { Account, Execution } from '../../types'
+import {
+  PublicClient,
+  encodeAbiParameters,
+  encodeFunctionData,
+  parseAbi,
+  slice,
+} from 'viem'
+import { isModuleInstalled } from './isModuleInstalled'
+import { Module, moduleTypeIds } from '../../../module/types'
+import { accountAbi } from '../constants/abis'
+
+export const installModule = ({
+  client,
+  account,
+  module,
+}: {
+  client: PublicClient
+  account: Account
+  module: Module
+}): Promise<Execution[]> => {
+  switch (module.type) {
+    case 'validator':
+    case 'executor':
+    case 'hook':
+      return _installModule({ client, account, module })
```

Comment:
I think that for executors, validators and fallbacks, the kernel encodes additional data into the initcode

Reply:
@kopy-kat additionalContext is needed for checking if module installed but we dont pass it for onChain installModule function